packages/api/src/models/study.ts

```diff
@@ -36,6 +41,18 @@ const studySchema = new Schema<IStudy>({
   description: { type: String, default: "" },
   batteries: [{ type: Schema.Types.ObjectId, ref: "CustomizedBattery" }],
   sessions: [sessionSchema],
+  serverCode: { type: String, unique: true },
 });
 
+studySchema.pre("save", async function (next) {
+  if (!this.serverCode) {
+    this.serverCode = uid.rnd(serverCodeLength);
+
+    const study = await Study.findOne({ serverCode: this.serverCode });
+    if (study) {
+      this.serverCode = uid.rnd(serverCodeLength);
```

Comment:
good approach, but this will need to be a while loop of some kind in case the regenerated code is also taken